lib/scala/pkg/src/main/scala/org/enso/pkg/ConfigCodecs.scala

```diff
@@ -0,0 +1,69 @@
+package org.enso.pkg
+
+import io.circe._
+import io.circe.syntax._
+
+/** A collection of utility codecs used in the [[Config]]. */
+object ConfigCodecs {
+
+  /** The common decoding failure.
+    *
+    * @param entity the name of decoded entity
+    * @param history the list of JSON cursor operations
+    */
+  private def decodingFailure(
+    entity: String,
+    history: List[CursorOp]
+  ): DecodingFailure =
+    DecodingFailure(s"Failed to decode $entity", history)
+
+  /** Try to decode the entity `A` from a JSON object.
+    *
+    * @param entity the name of decoded entity
+    * @param nameKey the name key of the JSON object
+    * @param cursor the current focus in the JSON document
+    */
+  def getFromObject[A: Decoder](
+    entity: String,
+    nameKey: String,
+    cursor: HCursor
+  ): Decoder.Result[A] =
+    cursor.keys match {
+      case Some(keys) if keys.nonEmpty =>
+        cursor
+          .get[A](nameKey)
+          .orElse {
+            cursor.get[Json](keys.head).flatMap { json =>
+              if (json.isNull) {
+                Decoder[A].decodeJson(keys.head.asJson)
+              } else {
+                Left(decodingFailure(entity, cursor.history))
+              }
+            }
```

Comment:
If I understand correctly, this is done so that both:
```
- module: foo
  property1: bar
- foo:
  property1: bar
```
deserialize to `Module(foo, property1->bar)` or something like that, right?
Still it was pretty hard to understand the intent behind this function, I had to read the code very carefully to get it.
I would suggest, if possible, adding a bit more of an explanation to the functions doc, so that it would be easier to grasp what is the purpose of this function.

Reply:
Right, I updated the doc